src/commonMain/kotlin/com/algolia/search/model/rule/Promotion.kt

```diff
@@ -1,19 +1,57 @@
+@file:Suppress("FunctionName")
+
 package com.algolia.search.model.rule
 
 import com.algolia.search.model.ObjectID
 import com.algolia.search.serialize.KeyObjectID
+import com.algolia.search.serialize.KeyObjectIDs
 import com.algolia.search.serialize.KeyPosition
 import kotlinx.serialization.SerialName
 import kotlinx.serialization.Serializable
 
 @Serializable
-public data class Promotion(
-    /**
-     * Unique identifier of the object to promote.
-     */
-    @SerialName(KeyObjectID) val objectID: ObjectID,
-    /**
-     * Promoted rank for the object.
-     */
-    @SerialName(KeyPosition) val position: Int
-)
+public sealed class Promotion {
+
+    abstract val position: Int
```

Comment:
I'm curious about this: if you put the Kdoc here, are they automatically reflected in the "child" classes below?

Reply:
Yes it is reflected in the child class unless overridden, we can do it here, however, we need to change the wording.